src/main/scala/lms/thirdparty/cuda/cudnn.scala

```diff
@@ -35,6 +35,94 @@ object CUDNNTypeLess extends Dsl with CLibs {
     assert(result.t == manifest[CUDNN_RESULT], "CUDNN_CHECK must take the CUDNN_RESULT type as input")
     UNIT(LIB_FUNCTION(manifest[Unit], "CUDNNCHECK", result.x)(Seq[Int](), Seq[Int](), Set[Int](), Adapter.CTRL))
   }
+
+  class CUDNN_TENSOR_DESCRIPTOR(override val x: Backend.Exp) extends TOP(x)
+  class CUDNN_FILTER_DESCRIPTOR(override val x: Backend.Exp) extends TOP(x)
+  class CUDNN_CONV_DESCRIPTOR(override val x: Backend.Exp) extends TOP(x)
+
+  
+  class CUDNN_TENSOR_FORMAT(override val x: Backend.Exp) extends TOP(x)
+  def CUDNN_NCHW(implicit __pos: SourceContext) = CMACRO("CUDNN_TENSOR_NCHW", manifest[Int])
+  def CUDNN_NHWC(implicit __pos: SourceContext) = CMACRO("CUDNN_TENSOR_NHWC", manifest[Int])
```

Comment:
I am wondering if `class CUDNN_TENSOR_FORMAT` definition is necessary, since the values of that type are just the `def CUDNN_NCHW` and `def CUDNN_NHWC`, which are `INT` instead.

Reply:
Yeah they are not necessary, I will change to `def ...():INT = INT(CMACRO(...))`, similar in cuda.scala